[No QA] 17885 custom user mentions renderer

robertKozik opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
This PR adds custom renderer used for user mentions inside chats

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/17885
PROPOSAL:https://docs.google.com/document/d/1vhvpwQDStA0lcsI0Z33pXOKPM6nXGRuplvQTFBWvCyA/edit#heading=h.5lyudapet5j6


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
Prerequisites: [ExpensiMark changes](https://github.com/Expensify/expensify-common/pull/524) are applied locally and app is in offline mode
1. open chat
2. type the message user mention ( its '@' followed with primary login - mail of the user) and send it
3. Properly rendered mention should appear in chat message with blue-ish background
4. type the message which mention your account
5. Properly rendered mention should appear in chat message with green-ish background

- [X] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
same as above
### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
6. Upload an image via copy paste
7. Verify a modal appears displaying a preview of that image
--->
same as above

In addition to mention testing, we need to check that code is rendering in chat properly, to check that there is no regression problems:
1. open chat
2. type the message with code block (code is the fragment between backticks f.e. \`code block\`) and send it
3. Properly rendered code block with rounded corners and horizontal padding should appear in chat message

- [X] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
![image](https://user-images.githubusercontent.com/61146594/235236988-1599168b-96df-483b-85dc-8ed973c9482e.png)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
<img width="371" alt="image" src="https://user-images.githubusercontent.com/61146594/235965533-de649ed3-a023-48c2-b670-a34957f62be0.png">


</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
<img width="404" alt="image" src="https://user-images.githubusercontent.com/61146594/235965197-1f60aca4-4a9a-4e7b-9e9d-37bbab421eb4.png">


</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
<img width="811" alt="image" src="https://user-images.githubusercontent.com/61146594/235240342-4e6c3a06-c60b-4616-91ba-6c8bb2a61fca.png">
<img width="817" alt="image" src="https://user-images.githubusercontent.com/61146594/235240377-7173cf0c-1612-4107-aedf-bfd5bc5af57f.png">

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
<img width="376" alt="image" src="https://user-images.githubusercontent.com/61146594/235241312-adb63d2e-95ef-4701-bbc9-b60ba5985ad2.png">

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
<img width="377" alt="image" src="https://user-images.githubusercontent.com/61146594/235960539-75119ee9-8906-4a02-85b1-b803cb0a5658.png">


</details>

robertKozik committed:
create styles for mentions

robertKozik committed:
create custom rendere for user mentions

robertKozik committed:
register custom user mentions renderer

puneetlath commented on src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js:
```diff
@@ -0,0 +1,44 @@
+            <Text
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...defaultRendererProps}
+                color={themeColors.textSupporting}
```
This should use `mentionText` or `ourMentionText` right?

puneetlath commented on src/components/HTMLEngineProvider/HTMLRenderers/index.js:
```diff
@@ -16,4 +17,5 @@ export default {
     // Custom tag renderers
     edited: EditedRenderer,
     pre: PreRenderer,
+    'mention-user': MentionUserRenderer,
```
We will have others, such as mention-here and mention-room.

puneetlath commented on src/styles/themes/default.js:
```diff
@@ -69,6 +69,10 @@ const darkTheme = {
+    mentionText: colors.blue200,
+    mentionBG: colors.blue700,
+    ourMentionText: colors.green200,
+    ourMentionBG: colors.green700,
```
In the design doc we said these should be: 
```
mentionText: colors.blue100
mentionBG: colors.blue600
ourMentionText: colors.green100
ourMentionBG: colors.green600
```
In case those aren't defined: 

- blue100: #BOD9FF
- blue600: #0164BF
- green100: #B1F2D6
- green600: #008C59

puneetlath reviewed (changes requested):
Just some minor comments. Overall looking really good!

shawnborton commented:
Looks like the corners are not round on iOS:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/2319350/235978136-1fc4dccd-edee-4c2c-a2e0-92148d32a91b.png">


Also, can we get more screenshots here that aren't just all offline mode?

shawnborton commented:
It also seems like the baseline of the text inside of the mention doesn't align to the baseline of the text outside of the mention. Ideally they should line up perfectly.

puneetlath commented:
@robertKozik you shouldn't need to test in offline mode anymore. The update to make it so that we don't strip the `<mention-user>` html tags is on staging.

puneetlath commented:
@shawnborton here's a screenshot from web from my own testing in non-offline mode. 

<img width="346" alt="image" src="https://user-images.githubusercontent.com/1371996/235979841-27b4a987-8bc0-41a8-a38f-a14a104fb708.png">

puneetlath commented:
@parasharrajat can you include a video of what you're seeing?

puneetlath commented:
@parasharrajat also, are you making sure to use staging server.

puneetlath commented:
>Should this be No QA
Good call

>I missed using the staging server. Can we please add that step to the Tests?
Yes, sounds good. Whenever the next back-end deploy happens (hopefully today) you'll be able to test this on staging or production

>Now I can't tag myself.
Ah interesting. @getusha it looks like our regex doesn't like email addresses with periods in them

puneetlath commented:
@getusha maybe we should just use the [isValidEmail regex](https://github.com/Expensify/expensify-common/blob/2b07f6ef7ac9cddba0e972b11b6bf07f0d9351c3/lib/str.js#L397) since it should already account for these cases.

robertKozik committed:
hoist mention styling to styleUtils file

robertKozik committed:
add JsDoc comment for mention navigation function

robertKozik committed:
add currentUserPresonalDetails for userMentionRenderer propTypes

puneetlath commented:
Awesome, thanks!

robertKozik committed:
define new colors for user mentions

robertKozik commented:
I've addressed all changes you requested. Styles related logic is passed to the styleUtils, props are filled, colors are redefined (100 and 600 values). Let me know if I missed something.

puneetlath commented:
@robertKozik I just wanted to double-check whether the issue on Android where the radius wasn't being applied to the background is resolved.

robertKozik commented:
On it 👀

robertKozik commented:
about the borderRadius problem on native side: https://github.com/Expensify/App/issues/17885#issuecomment-1534355869

robertKozik commented:
What you preffer. I see two different scenarios:
1. Removing radius and padding and create follow-up issue as @parasharrajat suggested
2. Use the workaround implemented for `code` and get things done here too. Queue up `HereRenderer` and `UserRenderer` to be refactored along side `CodeRenderer` 

@puneetlath I've never done it yet, but sure I'll test it. Use-case wise PR you linked should work perfectly for this one.

puneetlath commented:
>Use the workaround implemented for code and get things done here too. Queue up HereRenderer and UserRenderer to be refactored along side CodeRenderer

Let's do this one, since we are already using it elsewhere. And then we can create a follow-up issue to remove it once the RN update is merged.

puneetlath commented:
Awesome!

puneetlath commented:
Also, just to add some context, normally I wouldn't advocate doing a workaround. But I think it's okay this time because:
- we are trying to get mentions out ahead of an event mid-may
- this workaround already exists in our codebase
- we're already actively working on the fix that will allow us to get rid of the workaround

robertKozik commented:
How it looks with workaround:
<img width="332" alt="image" src="https://user-images.githubusercontent.com/61146594/236278030-b46efbd0-8a57-4085-a430-2285129d9c67.png">
In order get it I needed to change the font size to `13`  to be in line with code styling

robertKozik committed:
fix typo inside blue100 color value

robertKozik committed:
use workaround from CodeRenderer.js to render mentions with borderRadius on native devices

puneetlath commented:
I think that's fine for now. @shawnborton what do you think?

shawnborton commented:
Sorry, but what does it look like with font size 15? And this is just a temporary workaround, right? 

If we go with 13, can you slightly reduce the amount of horizontal padding too?

robertKozik commented:
There is not much room to adjust if we want to not change how `code` tag displays. With fontsize 15 we would encounter overflow:
<img width="289" alt="image" src="https://user-images.githubusercontent.com/61146594/236289536-e1e398bc-a30e-4738-a9f2-79ead88097b7.png">

Changing the padding will include some changes, so we will have to test `code` display too, but looks better:
<img width="283" alt="image" src="https://user-images.githubusercontent.com/61146594/236290537-64e0fb17-aadb-4b40-9ab7-3aee1c17a0b2.png">

shawnborton commented:
Okay, let's use 13 for now and then fix when we get the upstream changes in place.

robertKozik commented:
Rerequested review as new changes are already there. Later I'll update PR screenshots and QA test route (to cover code tag rendering as we were touching `WrappedText` Component)

robertKozik commented:
It looks like all these bugs are related to mention inside the link. I'm wondering if we want to parse mention when it is inside the link though? As such a mention can be only an coincidence inside the link from external site?

robertKozik commented on src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js:
```diff
@@ -0,0 +1,50 @@
+            <Text onPress={() => showUserDetails(loginWhithoutLeadingAt)}>
+
+                {/* We are using here workaround from CodeRenderer.js in order to apply borderRadius and padding for nested Text */}
+                <InlineCodeBlock
```
I preferred using the `InlineCodeBlock` as it is already covering the web and native implementations in the separate files. When we would be using `WrapperText` we should implement `MentionUserRenderer` separately for native and web, as WrapperText for web is overkill, when normal `Text` is working.

puneetlath commented:
Good catches @parasharrajat. I agree, all of those bugs sound like they are related to Expensimark and they should be fixed there. We shouldn't be putting mentions inside links or code blocks. So we can address those in https://github.com/Expensify/expensify-common/pull/524. 

Any bugs related to this PR specifically?

robertKozik commented:
Sure, I'm on it right now

robertKozik committed:
move propTypes to the top of the MentionUserRenderer file

robertKozik committed:
use font size from variables file, add missing padding for web mentions

robertKozik commented:
@parasharrajat Done - all changes you requested are implemented. I also saw on your screen that I removed padding for web mentions by mistake, I fixed it too.

robertKozik committed:
Update JsDoc comment

Update JsDoc comment for function inside MentionUserRenderer

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

robertKozik committed:
Update JsDoc comment

Update JsDoc comment for MentionUserRenderer propTypes

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

puneetlath commented:
Code looks good to me. So if the checklist tests well for you @parasharrajat I think we'll be good to merge.

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

cc: @puneetlath 

:ribbon: :eyes: :ribbon:  C+ reviewed

puneetlath commented:
Awesome!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

puneetlath commented:
What do you mean?

puneetlath commented:
Ah, damn!

puneetlath commented:
PR to revert: https://github.com/Expensify/App/pull/18490

robertKozik commented:
That escalated quickly 😮  I don't know how that happened, but sorry for that all reverting problem 😢

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4896622744) to staging by https://github.com/puneetlath in version: 1.3.11-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4922047640) to production by https://github.com/roryabraham in version: 1.3.12-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
